robottelo/ui/contentviews.py

```diff
@@ -379,103 +280,51 @@ def add_puppet_module(self, cv_name, module_name, filter_term):
 
         """
         element = self.search(cv_name)
-
-        if element:
-            element.click()
-            self.wait_for_ajax()
-            if self.wait_until_element(
-                    tab_locators["contentviews.tab_puppet_modules"]):
-                self.wait_until_element(
-                    tab_locators["contentviews.tab_puppet_modules"]).click()
-            else:
-                raise UIError(
-                    'Could not find tab to add puppet_modules'
-                )
-            self.wait_until_element(
-                locators["contentviews.add_module"]).click()
-            self.wait_for_ajax()
-            self.text_field_update(common_locators["cv_filter"], module_name)
-            strategy, value = locators["contentviews.select_module"]
-            element = self.wait_until_element((strategy, value % module_name))
-            if element:
-                element.click()
-                self.wait_for_ajax()
-                self.text_field_update(
-                    common_locators["cv_filter"], filter_term)
-                strategy, value = locators["contentviews.select_module_ver"]
-                element = self.wait_until_element(
-                    (strategy, value % filter_term))
-                if element:
-                    element.click()
-                    self.wait_for_ajax()
-                else:
-                    raise UIError(
-                        'Could not find the selected version "{0}" of puppet '
-                        'module "{1}"'.format(filter_term, module_name)
-                    )
-            else:
-                raise UIError(
-                    'Could not find the given puppet module "{0}"'
-                    .format(module_name)
-                )
-        else:
+        if not element:
             raise UIError(
                 'Could not find the selected CV "{0}"'.format(cv_name)
             )
 
+        element.click()
+        self.wait_for_ajax()
```

Comment:
self.click(locator) is out helper method `ui/base.py` which is working with locators only and doesn't have reference to `WebElement` entity
element.click() is a WebDriver method that is working with elements instead.

We created our helper method to get rid all `self.wait_until_element(locators["xxxxxxxxx"]).click()` or `self.find_element(locators["xxxxxxxxx"]).click()` statements with redundant supporting code.
To improve `element.click()` things we will need to create completely another method. But, it is another story:)